tests/full-example.ds

```diff
@@ -0,0 +1,42 @@
+extern {
+  fn format(x: str) : str;
+  fn printf(x: str);
+  fn sqrt(x: f64) : f64;
+}
+
+struct vec3 {
+  x: f32,
+  y: f32,
+  z: f32
+}
+
+impl vec3 {
+  fn length() : f32 {
+    f32(sqrt(self.x*self.x + self.y*self.y + self.z*self.z));
+  }
+  
+  fn to_str() : str {
+    format("Vec3 {X: %d, Y: %d, Z: %d}", self.x, self.y, self.z);
+  }
+}
+
+infix fn +(a: vec3, b: vec3) {
```

Comment:
missing return type

Reply:
In the example put in chat, there wasn't one.